roles/parse_operator_metadata/tasks/main.yml

```diff
@@ -1,109 +1,151 @@
+    set_fact:
+      csv_path: "{{ current_csv_path }}"
+      current_csv_dir: "{{ current_csv_path | dirname }}"
+
+  - name: "Determine the package path in the operator metadata directory"
+    find:
+      paths: "{{ current_csv_dir }}"
+      file_type: file
+      contains: 'kind: CustomResourceDefinition'
+    register: crd_paths_result
+
+  - name: "Set crd_paths to collect crd_paths"
+    set_fact:
+      crd_paths: []
+
+  - name: "Get paths from crd_paths_result"
+    set_fact:
+      crd_paths: "{{ crd_paths + [item.path] }}"
+    with_items: "{{ crd_paths_result['files'] }}"
+
+  - name: "Include the package path into the ansible vars"
+    include_vars:
+      file: "{{ csv_path }}"
+      name: operator_vars
+
+  - name: "Determine and setfact for podname"
+    set_fact:
+      operator_pod_name: "{{ operator_vars.spec.install.spec.deployments[0].name }}"
+      operator_container_name: "{{ operator_vars.spec.install.spec.deployments[0].spec.template.spec.containers[0].name }}"
+      operator_capabilities: "{{ operator_vars.metadata.annotations.capabilities }}"
+
+  - name: "Determine operator_allnamespaces_support"
+    set_fact:
+      operator_allnamespaces_support: "{{ operator_vars.spec.installModes | json_query(query) }}"
+    vars:
+      query: "[?type=='AllNamespaces'].supported"
+
+  - name: "Determine operator_ownnamespaces_support"
+    set_fact:
+      operator_ownnamespace_support: "{{ operator_vars.spec.installModes | json_query(query) }}"
+    vars:
+      query: "[?type=='OwnNamespaces'].supported"
+
+  - name: "Determine operator_ownnamespaces_support"
+    set_fact:
+      operator_singlenamespace_support: "{{ operator_vars.spec.installModes | json_query(query) }}"
+    vars:
+      query: "[?type=='SingleNamespaces'].supported"
+
+  - name: "Determine operator_ownnamespaces_support"
+    set_fact:
+      operator_multinamespace_support: "{{ operator_vars.spec.installModes | json_query(query) }}"
+    vars:
+      query: "[?type=='MultiNamespaces'].supported"
+
+  - name: "Set boolean value for different types of namespaces"
+    set_fact:
+      operator_allnamespaces_support: "{{ false if operator_allnamespaces_support == [] else operator_allnamespaces_support[0] }}"
+      operator_ownnamespace_support: "{{ false if operator_ownnamespace_support == [] else operator_ownnamespace_support[0] }}"
+      operator_singlenamespace_support: "{{ false if operator_singlenamespace_support == [] else operator_singlenamespace_support[0] }}"
+      operator_multinamespace_support: "{{ false if operator_multinamespace_support == [] else operator_singlenamespace_support[0] }}"
+
+  - name: "Output all collected data to a yaml file in work dir"
+    template:
+      src: "parsed_operator_data.yml.j2"
+      dest: "{{ work_dir }}/parsed_operator_data.yml"
+
+  rescue:
+
+    - name: "Rescue block contains the error messages"
+      debug:
+        msg: "We caught an error, can do stuff here to fix it, :-)"
```

Comment:
Are we expecting to improve this in this PR?

Reply:
@scoheb : The rescue block contains all the debugs statments required, 
including which task has failed. 
what is the failed task output.
Inorder to understand easily i have added specific names to all tasks for end users to understand what task has failed.

Reply:
@scoheb Are they any other details missing ?